pkg/groupsync/manytomanysyncer.go

```diff
@@ -83,6 +91,11 @@ func (f *ManyToManySyncer) Sync(ctx context.Context, sourceGroupID string) error
 		// get all source group IDs associated with the current target GroupID
 		sourceGroupIDs, err := f.targetGroupMapper.MappedGroupIDs(ctx, targetGroupID)
 		if err != nil {
+			logger.WarnContext(ctx, "error getting one ore more source group IDs for target group ID",
```

Comment:
Prefer "failed to" instead of "error doing"

Reply:
done.